lsp.go

```diff
@@ -346,26 +363,49 @@ func checkFile(fname string, content string) error {
 		if strings.HasSuffix(filename, ".tm") || strings.HasSuffix(filename, ".tm.hcl") {
 			path := filepath.Join(dir, filename)
 
-			if path == fname {
-				// file already added
+			if path == fromFile {
+				// ignore source file
 				continue
 			}
 
-			contents, err := os.ReadFile(path)
-			if err != nil {
-				log.Error().Err(err).Send()
-				return err
-			}
+			files = append(files, path)
+		}
+	}
 
-			err = parser.AddFile(path, contents)
-			if err != nil {
-				log.Error().Err(err).Send()
-				return err
-			}
+	return files, nil
+}
+
+// checkFiles checks if the given provided files have errors but the thisFile
+// is handled separately because it can be unsaved.
+func checkFiles(files []string, thisFile string, thisContent string) error {
```

Comment:
```suggestion
// checkFiles checks if the given provided files have errors but the unsavedFile
// is handled separately because it can be unsaved.
func checkFiles(files []string, unsavedFile string, unsavedContent string) error {
```
Not sure, just a suggestion.

Reply:
Or maybe "current" instead of "unsaved".

Reply:
I liked the `currentFile` and `currentContent` because sometimes it's saved, it's just the file that triggered the event. Thanks